session/auth/authprovider/authprovider.go

```diff
@@ -19,13 +20,21 @@ func NewDockerAuthProvider() session.Attachable {
 
 type authProvider struct {
 	config *configfile.ConfigFile
+
+	// The need for this mutex is not well understood.
```

Comment:
🤔 I don't see any code path that modifies `ap.config`

Reply:
This was not for protecting access to the config but the issue seems to be that the credentials function itself has problems when called in parallel. docker/cli#1862 has traces. @tiborvass possibly cli repo is a better place for this?